File: testdata/bench/kotlin/BenchmarkTest00021.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.http.*

// Vulnerable: XSS via unescaped user input in HTML response
suspend fun handler00021(call: ApplicationCall) {
    val name = call.parameters["name"]
    call.respondText("<html><body>Hello $name</body></html>", ContentType.Text.Html)
}

File: testdata/bench/javascript/BenchmarkTest00021.js
// Vulnerable: XSS via res.send with unsanitized user input
const express = require('express');
const router = express.Router();
router.get('/test021', (req, res) => {
    const name = req.query.name;
    res.send('<h1>Hello ' + name + '</h1>');
});
module.exports = router;
